src/middlewared/middlewared/plugins/docker_linux/images.py

```diff
@@ -0,0 +1,29 @@
+import os
+import subprocess
+
+from middlewared.service import CallError, private, CRUDService
+from middlewared.utils import Popen
+
+
+DEFAULT_DOCKER_IMAGES_PATH = '/usr/local/share/docker_images/docker-images.tar'
+
+
+class DockerImagesService(CRUDService):
+
+    class Config:
+        namespace = 'docker.images'
+
+    @private
+    async def load_images_from_file(self, path):
+        if not os.path.exists(path):
+            raise CallError(f'"{path}" path does not exist.')
+
+        # FIXME: Please do this in a better way
+        cp = await Popen(f'docker load < {path}', shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.PIPE)
```

Comment:
Doesnt the API support stdin so we dont have to use shell?

Reply:
It does for importing a single image, however I'll improve this - this was right now just done to have it working.